src/setup_payload/java/src/chip/setuppayload/SetupPayloadParser.java

```diff
@@ -69,7 +74,11 @@ private native SetupPayload fetchPayloadFromManualEntryCode(
       throws InvalidEntryCodeFormatException, SetupPayloadException;
 
   static {
-    System.loadLibrary("SetupPayloadParser");
+    try {
+      System.loadLibrary("SetupPayloadParser");
+    } catch (UnsatisfiedLinkError e) {
+      LOGGER.log(Level.SEVERE, "Cannot load library.", e);
```

Comment:
does this e.toString() include the name of the library that couldn't link? Might consider adding SetupPayloadParser to the log message if not.

Reply:
Can confirm it does include it:

10-21 10:07:43.890 29565 29843 E SetupPayloadParser: Cannot load library.
10-21 10:07:43.890 29565 29843 E SetupPayloadParser: java.lang.UnsatisfiedLinkError: dlopen failed: library "libSetupPayloadParser.so" not found
10-21 10:07:43.890 29565 29843 E SetupPayloadParser: 	at java.lang.Runtime.loadLibrary0(Runtime.java:1077)